dotnet/samples/PlayFabExamples/Example02_Generative/SegmentSkill.cs

```diff
@@ -17,6 +17,99 @@ namespace PlayFabExamples.Example02_Generative;
 /// </summary>
 public sealed class SegmentSkill
 {
+    /// <summary>
+    /// Create a segment using playfab open api json content.
+    /// </summary>
+    /// <param name="inputPrompt">Input prompt.</param>
+    /// <returns>Status of segment creation.</returns>
+    public async Task<string> CreateSegmentUsingOpenAPI(string inputPrompt)
+    {
+        // Step 1: Generate payload content to create a segment.
+        Console.WriteLine(inputPrompt);
+        var kernel = new KernelBuilder().WithLogger(ConsoleLogger.Logger)
+            .WithAzureChatCompletionService(TestConfiguration.AzureOpenAI.ChatDeploymentName, TestConfiguration.AzureOpenAI.Endpoint, TestConfiguration.AzureOpenAI.ApiKey, alsoAsTextCompletion: true, setAsDefault: true)
+            .Build();
+        SKContext context = kernel.CreateNewContext();
+        string miniJson = await GetMinifiedOpenApiJson();
+
+        string FunctionDefinition = @"
+You are an AI assistant for generating PlayFab input payload for given api. You have access to the full OpenAPI 3.0.1 specification.
+
+Api Spec:
+{{$apiSpec}}
+
+The CreateSegment operation in PlayFab Admin API requires a CreateSegmentRequest payload input.
+For FirstLoginFilter and LastLoginFilter, if the input value is days, convert value into minutes.
+Segment model name should be meaningful name from the input question.
+Don't provide any description about the answer. Only provide json payload content.
+Don't provide notes like below.
+Note: 30 days converted to minutes is 43200
+
+Example:
+Question: Create a segment for the players first logged in date greater than 2023-05-01?
+Answer: 
+{
+  ""SegmentModel"": {
+    ""Name"": ""FirstLoggedInPlayers"",
+    ""SegmentOrDefinitions"": [
+      {
+        ""SegmentAndDefinitions"": [
+          {
+            ""FirstLoginDateFilter"": {
+              ""LogInDate"": ""2023-05-01T00:00:00Z"",
+              ""Comparison"": ""GreaterThan""
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+
+Question:
+{{$input}}"
+.Replace("{{$apiSpec}}", miniJson, StringComparison.OrdinalIgnoreCase);
+
+        var playfabJsonFunction = kernel.CreateSemanticFunction(FunctionDefinition, temperature: 0.1, topP: 0.1);
```

Comment:
>var [](http://example.com/codeflow?start=8&length=3)

Here and for all the "var" below:
Specify the type explicitly

Only case not to specify it is when you do "new" (like in line 85